packages/validator/src/util/externalSignerClient.ts

```diff
@@ -160,17 +162,68 @@ export async function externalSignerUpCheck(remoteUrl: string): Promise<boolean>
     headers: {"Content-Type": "application/json"},
   });
 
-  const data = await handlerExternalSignerResponse<{status: string}>(res);
+  const data = await handlerExternalSignerUpCheckResponse(res);
   return data.status === "OK";
 }
 
-async function handlerExternalSignerResponse<T>(res: Response): Promise<T> {
+async function handlerExternalSignerGetKeysResponse<T>(res: Response, logger?: Logger): Promise<string[]> {
   if (!res.ok) {
     const errBody = await res.text();
+    logger?.debug("External signer GetKeys error", {body: errBody});
     throw Error(`${errBody}`);
   }
 
-  return JSON.parse(await res.text()) as T;
+  const resBody = await res.text();
+  const resContentType = res.headers.get("Content-Type");
+
+  logger?.debug("External signer GetKeys response", {contentType: resContentType, body: resBody});
```

Comment:
We generally try to avoid logging raw responses like this, there should already be a log in our validator store if validators are discovered, I think we even log all pubkeys on startup

Reply:
I had a hard time debugging the Diva remote signer as it failed with a generic JSON parser exception. I find it helpful to have this in the debug log, but if you generally avoid this I can remove it.

Reply:
> I find it helpful to have this in the debug log, but if you generally avoid this I can remove it.

I would have to run this myself to see what logs are repetitive

> I had a hard time debugging the Diva remote signer as it failed with a generic JSON parser exception

The problem here seems to be rather that the response handling made too many assumptions and zero validation on the response type and data.

If the response is properly validated and there are issues / inconsistencies you would get good error messages on what's wrong.

This is how it works in the generic http client as it uses ssz types to parse all data and would tell you if a property does match the expected format or is missing (we can improve that later here)

Reply:
Ok, I removed the log from externalSignerClient